### PR TITLE
Lock AWS provider version ">= 3.74, <5.0"

### PR DESCRIPTION
**Pull request by nin9swells**
## Summary
Related to #51
But instead of unlock the version restraint, expand the range of allowed version.

Ensure previous module version <v0.9.0 can be upgraded to >v0.9.0 even if they're already using AWS provider v4.

## How to reproduce error:
- Create stack with module v0.8.0 and using AWS provider v4
- Upgrade module to v0.9.0
- It will fail to upgrade, because of provider conflict

Error message
```
Could not retrieve the list of available versions for provider hashicorp/aws:
no available releases match the given constraints ~> 4.0, ~> 3.74
```

Terraform providers command output
```
$ terraform providers

Providers required by configuration:
.
├── provider[terraform.io/builtin/terraform]
├── provider[registry.terraform.io/hashicorp/aws] ~> 4.0
└── module.the_name_of_vpc
    ├── provider[registry.terraform.io/hashicorp/random] >= 1.2.*, < 3.0.0
    ├── provider[registry.terraform.io/hashicorp/aws] ~> 3.74
    └── module.flowlogs_to_s3_naming
        └── provider[registry.terraform.io/hashicorp/random]
```

## Change Logs
```
FIX:
- Unlock AWS provider version to ensure module version <0.9.0 can be upgraded even it's provisioned using AWS provider v4.

```

## Contributor Notes

Additional notes for future improvement:
- We might reconsider again to lock provider, but with higher version, not locking to lower version available on internet

### CODE REVIEW

**Review by vincenttjia — APPROVED** [review]

File: versions.tf [hunk 1]
```diff
@@ -2,6 +2,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
+      version = ">= 3.74, <5.0"
```
Comment (radengoots) [inline versions.tf:5]: I just remembered that this constraint defined is due to the S3 bucket resource still using the old v3 format. Thus, the update to the later version (v4) must include an update to v4 (separating the S3 bucket resource into multiple resources) and also release a non-backwards compatible version (v1.x, perhaps).

**Review by radengoots — CHANGES_REQUESTED** [review]

**Review by radengoots — CHANGES_REQUESTED** [review]
To support aws v4, the S3 bucket resources in the module need to be updated.

File: versions.tf (hunk 1 above)
Comment (nin9swells) [reply versions.tf:5]: I don't think S3 bucket resource will break on https://github.com/traveloka/terraform-aws-vpc/blob/master/main.tf#L669-L715

There are 3 argument there and all of them are just deprecated on v4:
- [server_side_encryption_configuration](https://github.com/traveloka/terraform-aws-vpc/blob/master/main.tf#L673-L679)
- [lifecycle_rule](https://github.com/traveloka/terraform-aws-vpc/blob/master/main.tf#L681-L693)
- [logging](https://github.com/traveloka/terraform-aws-vpc/blob/master/main.tf#L695-L698)

Comment (nin9swells) [reply versions.tf:5]: We can solve this on the next PR
while allowing other module user to proceed with upgrade first.

**Review by radengoots — APPROVED** [review]
LGTM

**Pull request merged**